pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java

```diff
@@ -237,11 +237,16 @@ private void reorderDimsAndBuildMap()
     _colNameToIntMap = new HashMap<>();
 
     _dimNamesInvertedSortedIndexApplicable = new HashSet<>(_dimNames);
+    _dimNamesInvertedSortedIndexApplicable.addAll(_metricNames);
+    _dimNamesInvertedSortedIndexApplicable.addAll(_dateTimeNames);
+
     _dimNamesInvertedSortedIndexApplicable.remove(sortedColumn);
     _dimNamesInvertedSortedIndexApplicable.removeAll(invertedIndexColumns);
     _dimNamesInvertedSortedIndexApplicable.removeAll(noDictionaryColumns);
 
     HashSet<String> dimNamesInveredSortedIndexNotApplicable = new HashSet<>(_dimNames);
+    dimNamesInveredSortedIndexNotApplicable.addAll(_metricNames);
```

Comment:
Why do we need `dimNamesInveredSortedIndexNotApplicable` ? I feel we can remove this and do some cleanup

Reply:
right, we can write it in clean way

Reply:
updated!